hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/ITTestHoodieDataSource.java

```diff
@@ -1835,6 +1885,75 @@ private static Stream<Arguments> indexAndTableTypeParams() {
     return Stream.of(data).map(Arguments::of);
   }
 
+  /**
+   * Return test params => (HoodieTableType, hive style partitioning, filter condition, expected result).
+   */
+  private static Stream<Arguments> tableTypeAndFilters() {
+    HoodieTableType[] tableTypes = {HoodieTableType.COPY_ON_WRITE, HoodieTableType.MERGE_ON_READ};
+    boolean [] isHiveStylePartitions = {true, false};
+    Pair<String,  List<RowData>>[] filterAndResults = new Pair[] {
+        Pair.of(
+            "`partition` = 'par5' or `partition` = 'par6'",
+            Arrays.asList(
+                insertRow(StringData.fromString("id9"), StringData.fromString("LiLi"), 24,
+                    TimestampData.fromEpochMillis(9), StringData.fromString("par5")))),
+        Pair.of(
+        "`partition` >= 'par5' and `partition` <= 'par6'",
+        Arrays.asList(
+            insertRow(StringData.fromString("id9"), StringData.fromString("LiLi"), 24,
+                TimestampData.fromEpochMillis(9), StringData.fromString("par5")))),
+        Pair.of(
+        "`partition` <> 'par7'",
+        Arrays.asList(
+            insertRow(StringData.fromString("id1"), StringData.fromString("Danny"), 23,
+                TimestampData.fromEpochMillis(1), StringData.fromString("par1")),
+            insertRow(StringData.fromString("id2"), StringData.fromString("Stephen"), 33,
+                TimestampData.fromEpochMillis(2), StringData.fromString("par1")),
+            insertRow(StringData.fromString("id3"), StringData.fromString("Julian"), 53,
```

Comment:
Can the test be simplified by using one input data set?

Reply:
Already use one input data set.
Those data is output data, which is different based on different filter conditions.